include/odp/api/spec/proto_stats.h

```diff
@@ -0,0 +1,205 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(C) 2021 Marvell.
+ */
+
+/**
+ * @file
+ *
+ * ODP Proto Stats
+ */
+
+#ifndef ODP_API_SPEC_PROTO_STATS_H_
+#define ODP_API_SPEC_PROTO_STATS_H_
+#include <odp/visibility_begin.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @addtogroup odp_proto_stats
+ *  @{
+ */
+
+/**
+ * @def ODP_PROTO_STATS_INVALID
+ * Invalid proto stats handle
+ */
+
+/** ODP proto stats id */
+typedef enum odp_proto_stats_id_t {
+	/** Tx proto stats id ODP_PROTO_STATS_ID_TX_*
+	 *
+	 * - Statistics that can be enabled in proto stats objects for Tx of a packet via
+	 *   Packet IO. Packet IO pktout config `odp_pktout_config_opt_t::bit::proto_stats_ena`
+	 *   needs to be enabled for the offload to work.
+	 *
+	 * - Tx packet and octet sent/drop statistics might include packets sent/dropped via
+	 *   Traffic Manager or Tx packet Aging or due to any other Tx errors. It is
+	 *   implementation specific as to what all Tx sent/drop events are accounted for.
+	 */
+
+	/** Packet sent count */
+	ODP_PROTO_STATS_ID_TX_PKT,
+	/** Packet drop count */
+	ODP_PROTO_STATS_ID_TX_PKT_DROP,
+	/** Packet sent Octet counter 0 */
+	ODP_PROTO_STATS_ID_TX_OCT_COUNT0,
+	/** Packet drop Octet counter 0 */
+	ODP_PROTO_STATS_ID_TX_OCT_COUNT0_DROP,
+	/** Packet sent octet counter 1 */
+	ODP_PROTO_STATS_ID_TX_OCT_COUNT1,
+	/** Packet drop octet counter 1 */
+	ODP_PROTO_STATS_ID_TX_OCT_COUNT1_DROP,
+	/** Max */
+	ODP_PROTO_STATS_ID_MAX,
+} odp_proto_stats_id_t;
+
+/** ODP proto stats params */
+typedef struct odp_proto_stats_param_t {
+	/** Stats id mask
+	 *
+	 * Each bit corresponds to one ODP_PROTO_STATS_ID_* and
+	 * bit position is same as ODP_PROTO_STATS_ID_*.
+	 * Bits set in this mask are the stat counters to be enabled
+	 * in the created proto stats object.
+	 */
+	uint64_t stats_mask;
+} odp_proto_stats_param_t;
```

Comment:
It would match better other stats APIs, mask can be build with a bit field or if enum values would be directly the bit positions:
mask.all = 0;
mask.tx_pkt =1;
mask.tx_pkt_drop = 1;
OR
mask = ODP_PROTO_STATS_ID_TX_PKT | ODP_PROTO_STATS_ID_TX_PKT_DROP;

CURRENTLY:
mask =  1 << ODP_PROTO_STATS_ID_TX_PKT | 1 << ODP_PROTO_STATS_ID_TX_PKT_DROP;

I think the current proposal is less user friendly than the other two. The bit field would match other APIs, but may not produce simplest implementation of the single counter read function.  On the other hand, how often a single counter would be read - could we just remove the single counter read function.

Reply:
Checked with Transport team. Read all seems to be sufficient. I'll change the mask to bitfield as you mentioned above.